src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -205,15 +205,16 @@ String getFilePath() {
   private CompletableFuture<Void> distributeFlush(boolean isForce, long timeDiffMillis) {
     return CompletableFuture.runAsync(
         () -> {
-          logger.logDebug(
-              "Submit flush task on client={}, isForce={}, isNeedFlush={}, timeDiffMillis={},"
-                  + " currentDiffMillis={}",
-              this.owningClient.getName(),
-              isForce,
-              this.isNeedFlush,
-              timeDiffMillis,
-              System.currentTimeMillis() - this.lastFlushTime);
-
+          if (this.isNeedFlush) {
```

Comment:
This will cause us to not debug log flushes triggered by channel closing and the minimum time interval.  Is there a specific issue you're running into here filtering out time based flushes?

Reply:
I think the only reason is the noisy nature/frequency of this. 
Many times when we ask customer to enable debug logging for debugging issues. 

In case of closing I think `isForce`= true, I can add a check if isForce is also true so if either of them is true, we should log. WDYT?

Reply:
I lean towards keeping it as it is.  This change would remove one debug log line every second which doesn't seem super noisy to me for a debug line

Reply:
I see one more reason is:
1. I deleted the connector and the API it invokes is same when the rebalance tries to close all partition. During KC partition close, lets say I dont need to close client but during connector close(Delete), I do need to close client. 
2. Not closing a client will continue to see this debug log which might be confusing to customers. 

One other work around is to close client also during rebalance..

Reply:
For a compromise, could we add a parameter that would allow power users to suppress this log line in the way you've outlined but leave the log as is by default?

Reply:
No objection on removing it, as it may not be very useful, the only thing we will miss is the ability to determine whether the worker has stopped or not

Reply:
This is how I have set the logger for snowflake ingest
log4j.logger.net.snowflake=DEBUG
We can also set at class level
log4j.logger.net.snowflake.ingest.streaming.internal.FlushService=DEBUG


Or on a completely different level, should I just trace it?

Reply:
I'm fine with dropping it to Trace but logging every flush